django_nitro_mailer/backend.py

```diff
@@ -18,3 +24,33 @@ def send_messages(self, email_messages: Iterable[EmailMessage]) -> int:
 
         email_db_list = Email.objects.bulk_create(email_list, batch_size=1000)
         return len(email_db_list)
+
+
+class SyncBackend(BaseEmailBackend):
+    def get_connection(self, fail_silently=False):
+        backend_path = settings.NITRO_MAILER_EMAIL_BACKEND
+        return django_get_connection(backend_path, fail_silently=fail_silently)
+
+    def send_messages(self, email_messages: Iterable[EmailMessage]) -> int:
+        successful_sends = 0
+        connection = self.get_connection()
+        for email_message in email_messages:
+            try:
+                connection.send_messages([email_message])
+                if email_db_logging:
+                    EmailLog.objects.create(
+                        email_data=email_message.message().as_bytes(), result=EmailLog.Results.SUCCESS
+                    )
+                logger.info(
+                    "Email sent successfully", extra={"recipients": email_message.to, "subject": email_message.subject}
+                )
+                successful_sends += 1
+            except Exception as e:
+                if email_db_logging:
+                    EmailLog.objects.create(
+                        email_data=email_message.message().as_bytes(), result=EmailLog.Results.FAILURE
+                    )
+                logger.error(f"Failed to send email: {e}", exc_info=True)
+            throttle_email_delivery()
+
+        return successful_sends
```

Comment:
Potrzebowałbym, zrobiła funkcję `send_email_message`  która by przyjmowała pojedynczą wiadomość `EmailMessage` i robiła to, co robi teraz `send_emails`. Następnie wykorzystywała tę funkcję zarówno tu jak i tam.